sources/packages/backend/apps/queue-consumers/src/services/student-file/student-file.service.ts

```diff
@@ -29,33 +35,55 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
    * for any viruses.
    * @param uniqueFileName unique filename of the file to perform the virus scan.
    * @param processSummary process summary logs.
-   * @returns boolean true if the file is virus infected, false otherwise.
+   * @returns boolean true if the file is virus infected, false not infected or null otherwise.
    */
   async scanFile(
     uniqueFileName: string,
     processSummary: ProcessSummary,
   ): Promise<boolean> {
     const studentFile = await this.getStudentFile(uniqueFileName);
     if (!studentFile) {
-      throw new NotFoundException(`Student file ${uniqueFileName} not found.`);
+      throw new CustomNamedError(
+        `File ${uniqueFileName} is not found or has already been scanned for viruses. Scanning the file for viruses is aborted.`,
+        FILE_NOT_FOUND,
+      );
     }
+
     const stream = new Readable();
     stream.push(studentFile.fileContent);
     stream.push(null);
-    const isInfected = await this.clamAVService.scanFile(
-      stream,
-      processSummary,
-    );
-    if (isInfected == null) {
-      // If the file could not be scanned for any reason,
-      // update the file scan status to Pending to keep a
-      // track of the files that haven't been scanned yet.
-      await this.updateFileScanStatus(studentFile.uniqueFileName, isInfected);
-      throw new CustomNamedError(
-        `Unable to scan the file ${studentFile.uniqueFileName}`,
-        UNABLE_TO_SCAN_FILE,
+    let isInfected: boolean;
+    let errorMessage = `Unable to scan the file ${uniqueFileName} for viruses.`;
+    let errorName = "";
+    try {
+      isInfected = await this.clamAVService.scanFile(stream);
```

Comment:
Since `isInfected` will no longer be undefined, the undefined can be removed from the switch case in the method `updateFileScanStatus` to avoid confusion.